### PR TITLE
Kill some old exports stuff

### PR DESCRIPTION
**Pull request by esoergel**
This is just a start, but it conflicts with what Jenny's been doing, and I figure these deletions will have to come in piecemeal.

No instances of `SavedExportSchema`, `HQExportSchema`, `FormExportSchema`, or `CaseExportSchema` exist in any environment, so this code begins by attempting to remove things that rely on them.  Commit-by-commit should make the most sense, I'll try to go through and annotate.

### CODE REVIEW

File: corehq/apps/export/views.py [hunk 1]
```diff
@@ -1167,36 +1165,23 @@ def create_export_form_title(self):
         return _("Select a Form to Export")
 
     def fmt_export_data(self, export):
-        if isinstance(export, FormExportSchema):
```
Comment (esoergel) [inline corehq/apps/export/views.py]: ^ operative change

File: corehq/apps/export/views.py [hunk 2]
```diff
@@ -1894,13 +1879,6 @@ class DownloadNewFormExportView(GenericDownloadNewExportMixin, BaseDownloadExpor
     check_for_multimedia = True
     form_or_case = 'form'
 
-    @staticmethod
-    def get_export_schema(domain, export_id):
```
Comment (esoergel) [inline corehq/apps/export/views.py]: This might actually belong with another commit, sorry about that.  Suffice to say that it gets deleted eventually either way.

File: corehq/apps/dashboard/models.py [hunk 3]
```diff
@@ -229,8 +229,6 @@ def _paginated_items(self, items_per_page, skip):
                 urlname = 'new_export_download_forms'
             elif isinstance(export, CaseExportSchema):
                 urlname = 'export_download_cases'
-            elif isinstance(export, FormExportSchema):
-                urlname = 'export_download_forms'
```
Comment (esoergel) [inline corehq/apps/dashboard/models.py]: Same as before - this is now unreferenced, so I can delete the view and merge its contents down to the subclass.

File: corehq/apps/export/views.py [hunk 4]
```diff
@@ -618,41 +617,6 @@ def _get_daily_saved_export_metadata(self, export):
             'taskStatus': self._get_task_status_json(export._id),
         }
 
-    def fmt_legacy_emailed_export_data(self, group_id=None, index=None,
```
Comment (esoergel) [inline corehq/apps/export/views.py]: this was only referenced the once

File: corehq/apps/export/views.py [hunk 5]
```diff
@@ -672,23 +636,6 @@ def _fmt_emailed_export_fileData(self, fileId, size, last_updated,
             'downloadUrl': download_url,
         }
 
-    def get_formatted_emailed_export(self, export):
```
Comment (esoergel) [inline corehq/apps/export/views.py]: this was only referenced once

File: corehq/apps/export/views.py [hunk 6]
```diff
@@ -618,41 +617,6 @@ def _get_daily_saved_export_metadata(self, export):
...
-        """
-        Return a dictionary containing details about an emailed export.
-        This will eventually be passed to an Angular controller.
-        """
-        file_data = {}
-        if has_file:
-            if is_safe:
-                saved_download_url = 'hq_deid_download_saved_export'
-            else:
-                saved_download_url = 'hq_download_saved_export'
```
Comment (esoergel) [inline corehq/apps/export/views.py]: These two views can also be removed now, will do that in another PR

File: corehq/apps/export/views.py [hunk 7]
```diff
@@ -325,7 +318,7 @@ def export_list(self):
         return exports
 
     def _get_export(self, domain, export_id):
-        return self.get_export_schema(self.domain, export_id)
+        raise NotImplementedError()
```
Comment (esoergel) [inline corehq/apps/export/views.py]: This is now overridden in all child classes

File: corehq/apps/export/views.py [hunk 8]
```diff
@@ -403,9 +396,8 @@ def _get_download_task(self, in_data):
         return download
 
     def _get_and_rebuild_export_schema(self, export_id):
-        export_object = self.get_export_schema(self.domain, export_id)
-        export_object.update_schema()
-        return export_object
+        # TODO
+        raise Exception("this must be failing if it's ever called...")
```
Comment (esoergel) [inline corehq/apps/export/views.py]: it's not - deleted in a later commit

File: corehq/apps/export/views.py [hunk 9]
```diff
@@ -2016,10 +2008,6 @@ class DownloadNewSmsExportView(GenericDownloadNewExportMixin, BaseDownloadExport
     export_id = None
     sms_export = True
 
-    @staticmethod
-    def get_export_schema(domain, include_metadata):
-        return SMSExportDataSchema.get_latest_export_schema(domain, include_metadata)
```
Comment (esoergel) [inline corehq/apps/export/views.py]: I don't know what the deal with this method was - none of the callers match this method signature.

File: corehq/apps/export/views.py [hunk 10]
```diff
@@ -385,15 +394,15 @@ def poll_custom_export_download(self, in_data):
         return context
 
     def _get_download_task(self, in_data):
```
Comment (esoergel) [inline corehq/apps/export/views.py]: I copied the `GenericDownloadNewExportMixin` version over the original

File: corehq/apps/export/views.py [hunk 11]
```diff
@@ -446,8 +455,11 @@ def _process_filters_and_specs(self, in_data):
         """Returns a the export filters and a list of JSON export specs
         """
```
Comment (esoergel) [inline corehq/apps/export/views.py]: Similarly, I copied the GenericDownloadNewExportMixin version over the original

File: corehq/apps/export/views.py [hunk 12]
```diff
@@ -404,37 +403,6 @@ def _get_download_task(self, in_data):
             filename=self._get_filename(export_instances)
         )
 
-    def _get_and_rebuild_export_schema(self, export_id):
-        # TODO
-        raise Exception("this must be failing if it's ever called...")
-
-    def _get_bulk_download_task(self, export_specs, export_filter):
```
Comment (esoergel) [inline corehq/apps/export/views.py]: References to this method and the one below it were both removed when I overwrote `_get_download_task` in the previous commit.

File: corehq/apps/export/forms.py [hunk 13]
```diff
@@ -393,11 +392,6 @@ def _get_es_user_types(self, mobile_user_and_groups_slugs=None):
             es_user_types.extend(export_to_es_user_types_map[type_])
         return es_user_types
 
-    def _get_group(self):
```
Comment (esoergel) [inline corehq/apps/export/forms.py]: This was only referenced in the classes I'm removing

File: corehq/apps/export/forms.py [hunk 14]
```diff
@@ -1203,7 +1144,7 @@ class FilterCaseESExportDownloadForm(EmwfFilterExportMixin, GenericFilterCaseExp
     def __init__(self, domain_object, timezone, *args, **kwargs):
         self.timezone = timezone
         self.skip_layout = True
-        super(FilterCaseESExportDownloadForm, self).__init__(domain_object, timezone, *args, **kwargs)
+        super(FilterCaseESExportDownloadForm, self).__init__(domain_object, *args, **kwargs)
```
Comment (esoergel) [inline corehq/apps/export/forms.py]: This change mimics the functionality of the now defunct `GenericFilterCaseExportDownloadForm`

File: corehq/apps/export/views.py [hunk 15]
```diff
@@ -1406,16 +1379,12 @@ def fmt_export_data(self, export):
             'emailedExport': emailed_export,
             'editUrl': reverse(EditNewCustomFormExportView.urlname,
                                args=(self.domain, export.get_id)),
-            'downloadUrl': self._get_download_url(export.get_id, is_legacy),
+            'downloadUrl': self._get_download_url(export.get_id),
```
Comment (orangejenny) [inline corehq/apps/export/views.py]: Stylistic comment: I'd get rid of `_get_download_url` altogether and just inline the `reverse` call.

File: corehq/apps/export/views.py (hunk 9 above)
Comment (orangejenny) [reply corehq/apps/export/views.py]: Odd, even the [original implementation](https://github.com/dimagi/commcare-hq/pull/14823/files#diff-1086f45b5d86ddf7a9cd7b848caf29a6R2298) doesn't have any callers that match.

**Review by orangejenny — APPROVED** [review]
